### DISCUSSION TITLE
Router routeChangeComplete not getting called

### DISCUSSION THREAD

**Discussion by dtanaka1234** [vercel/next.js#25813] [Help]
I can't seem to get the nextjs router event `routeChangeComplete ` to be called when the route changes. My code is pretty much identical to what's in the documentation:

`
   const router = useRouter();

    useEffect(() => {
        function handleRouteChange() {
             console.log();
        }

        router.events.on('routeChangeComplete', handleRouteChange);
        return () => {
            router.events.off('routeChangeComplete', handleRouteChange);
        };
    }, [router]);
`

The only different is that this is in the component for that route and not in the global app context. `routeChangeComplete` only seems to be working in the global context. Is there something I'm missing here when trying to see if the route changed inside of my component (which is part of the page props component)?

**Comment by leerob** [comment]
Here's an example looking at `routeChangeComplete`.

```js
import { useRouter } from 'next/router';
import { useEffect } from 'react';

export default function RouteHook() {
  const router = useRouter();

  useEffect(() => {
    const start = () => {
      // Do something
    };

    const done = () => {
      // Do something
    };

    router.events.on('routeChangeStart', start);
    router.events.on('routeChangeComplete', done);
    router.events.on('routeChangeError', done);
    
    return () => {
      router.events.off('routeChangeStart', start);
      router.events.off('routeChangeComplete', done);
      router.events.off('routeChangeError', done);
    };
    
  }, []);
  
  return <></>;
}
```

**Reply by dtanaka1234** [reply]
Hi Lee,

Thanks for the response! I put this code into the project however `routeChangeComplete ` is still not getting called when changing routes in the nextjs app. I think the issue might be that this hook is an component inside the component that is being loaded (from the url  route change). `routeChangeComplete ` works fine if this event handler is created in the global context. Are you able to get this to work with a component being loaded with the route change?
